static/src/javascripts/projects/common/modules/navigation/new-header.js

```diff
@@ -178,58 +177,63 @@ const toggleMenu = (): void => {
         } else {
             focusFirstMenuSection();
-
-            if (!avatarIsEnhanced) {
-                enhanceAvatar();
-                avatarIsEnhanced = true;
-            }
         }
     };
 
     fastdom.write(update);
 };
 
-const toggleEditionPickerDropdown = () => {
+const toggleDropdown = (menuAndTriggerEls: Object) => {
     const openClass = 'dropdown-menu--open';
 
-    fastdom
-        .read(() => ({
-            menu: document.querySelector('.js-edition-dropdown-menu'),
-            trigger: document.querySelector('.js-edition-picker-trigger'),
-        }))
-        .then(els => {
-            const { menu, trigger } = els;
+    fastdom.read(() => menuAndTriggerEls).then(els => {
+        const { menu, trigger } = els;
+
+        if (!menu) {
+            return;
+        }
 
-            if (!menu) {
-                return;
+        const isOpen = menu.classList.contains(openClass);
+        const expandedAttr = isOpen ? 'false' : 'true';
+        const hiddenAttr = isOpen ? 'true' : 'false';
+
+        return fastdom.write(() => {
+            if (trigger) {
+                trigger.setAttribute('aria-expanded', expandedAttr);
             }
 
-            const isOpen = menu.classList.contains(openClass);
-            const expandedAttr = isOpen ? 'false' : 'true';
-            const hiddenAttr = isOpen ? 'true' : 'false';
+            menu.setAttribute('aria-hidden', hiddenAttr);
+            menu.classList.toggle(openClass, !isOpen);
 
-            return fastdom.write(() => {
-                if (trigger) {
-                    trigger.setAttribute('aria-expanded', expandedAttr);
-                }
+            if (!isOpen) {
+                mediator.on('module:clickstream:click', clickSpec => {
+                    const elem = clickSpec ? clickSpec.target : null;
 
-                menu.setAttribute('aria-hidden', hiddenAttr);
-                menu.classList.toggle(openClass, !isOpen);
-
-                if (!isOpen) {
-                    mediator.on('module:clickstream:click', clickSpec => {
-                        const elem = clickSpec ? clickSpec.target : null;
-
-                        // if anywhere else but the links are clicked, the dropdown will close
-                        if (elem !== menu) {
-                            toggleEditionPickerDropdown();
-                        }
-                    });
-                } else {
-                    // when the dropdown closes, remove event
-                    mediator.removeEvent('module:clickstream:click');
-                }
-            });
+                    // if anywhere else but the links are clicked, the dropdown will close
+                    if (elem !== menu) {
+                        toggleDropdown(menuAndTriggerEls);
+                    }
+                });
+            } else {
+                // when the dropdown closes, remove event
+                mediator.removeEvent('module:clickstream:click');
```

Comment:
I know this is not something you've introduced, but it looks like this removes any event called `module:clickstream:click`, which is [used in a few different places](https://github.com/guardian/frontend/search?utf8=%E2%9C%93&q=on%28%27module%3Aclickstream%3Aclick%27&type=). It might be safer to name the callback passed to `mediator.on(...)` and [remove that specific event listener](https://github.com/Olical/EventEmitter/blob/master/docs/api.md#removelistener) here. What do you think?